lib/worker.js

```diff
@@ -0,0 +1,216 @@
+}
+
+const util = require('util');
+const assert = require('assert');
+const EventEmitter = require('events');
+const WorkerContextBinding = process.binding('worker_context');
+const JSONStringify = function(value) {
+  if (value === undefined) value = null;
+  return JSON.stringify(value);
+};
+const JSONParse = JSON.parse;
+const EMPTY_ARRAY = [];
+
+const workerContextSymbol = Symbol('workerContext');
+const installEventsSymbol = Symbol('installEvents');
+const checkAliveSymbol = Symbol('checkAlive');
+const initSymbol = WorkerContextBinding.initSymbol;
+
+const builtinErrorTypes = new Map([
+  Error, SyntaxError, RangeError, URIError, TypeError, EvalError, ReferenceError
+].map(function(Type) {
+  return [Type.name, Type];
+}));
+
+const Worker = WorkerContextBinding.JsConstructor;
+util.inherits(Worker, EventEmitter);
+
+Worker.prototype[initSymbol] = function(entryModulePath, options) {
+  if (typeof entryModulePath !== 'string')
+    throw new TypeError('entryModulePath must be a string');
+  EventEmitter.call(this);
+  options = Object(options);
+  const keepAlive =
+      options.keepAlive === undefined ? true : !!options.keepAlive;
+  const evalCode = !!options.eval;
+  const userData = JSONStringify(options.data);
+  this[workerContextSymbol] =
+      new WorkerContextBinding.WorkerContext(entryModulePath, {
+        keepAlive: keepAlive,
+        userData: userData,
+        eval: evalCode
+      });
+  this[installEventsSymbol]();
+};
+
+Worker.prototype[installEventsSymbol] = function() {
+  const workerObject = this;
+  const workerContext = this[workerContextSymbol];
+
+  const onerror = function(payload) {
+    var ErrorConstructor = builtinErrorTypes.get(payload.builtinType);
+    if (typeof ErrorConstructor !== 'function')
+      ErrorConstructor = Error;
+    const error = new ErrorConstructor(payload.message);
+    error.stack = payload.stack;
+    util._extend(error, payload.additionalProperties);
+    workerObject.emit('error', error);
+  };
+
+  workerContext._onexit = function(exitCode) {
+    workerObject[workerContextSymbol] = null;
+    workerObject.emit('exit', exitCode);
+  };
+
+  workerContext._onmessage = function(payload, messageType) {
+    payload = JSONParse(payload);
+    switch (messageType) {
+      case WorkerContextBinding.USER:
+        return workerObject.emit('message', payload);
+      case WorkerContextBinding.EXCEPTION:
+        return onerror(payload);
+      default:
+        assert.fail('unreachable');
+    }
+  };
+};
+
+Worker.prototype[checkAliveSymbol] = function() {
+  if (!this[workerContextSymbol])
+    throw new RangeError('this worker has been terminated');
+};
+
+Worker.prototype.postMessage = function(payload) {
+  this[checkAliveSymbol]();
+  this[workerContextSymbol].postMessage(JSONStringify(payload),
+                                        EMPTY_ARRAY,
+                                        WorkerContextBinding.USER);
+};
+
+Worker.prototype.terminate = function(callback) {
```

Comment:
@tflanagan depends on what you want to be closer to: node `child_process` or Web Workers.